### PR TITLE
small refactoring

### PR DESCRIPTION
**Pull request by anthony-redFox**

### CODE REVIEW

File: lib/middleware/common.js [hunk 1]
```diff
@@ -99,22 +99,7 @@ function setHeavyCacheHeaders (response) {
...
-    return this.promise.then(success, error)
-  }
-
-  set (newPromise) {
-    this.promise = newPromise
-  }
-}
-
 // PUBLIC API
-exports.PromiseContainer = PromiseContainer
```
Comment (johnjbarton) [inline lib/middleware/common.js]: I wonder if this could break users who are using this in plugins.

Can you put the delete in a separate PR and comment that it is deprecated in this PR?

Comment (anthony-redFox) [reply lib/middleware/common.js]: done

**Comment by anthony-redFox** [comment]
@johnjbarton cc

**Comment by johnjbarton** [comment]
Really sorry for the delay.  

We've switched to commitlint and semantic release.  That means merged commits will drive the release (yay). But the commits need to be describe in a formatted way.

Can you rebase and repush? That will force you through the commitlint.  I guess yours would be something like
chore(server): Convert PromiseContainer to object and remove initializeMimeTypes.

**Comment by anthony-redFox** [comment]
@johnjbarton  done
